ethminer/MinerAux.h

```diff
@@ -562,16 +605,19 @@ class MinerCLI
 			CLMiner::setThreadsPerHash(m_openclThreadsPerHash);
 
 			if (!CLMiner::configureGPU(
-					m_localWorkSize,
-					m_globalWorkSizeMultiplier,
-					m_openclPlatform,
-					0,
-					m_dagLoadMode,
-					m_dagCreateDevice,
-					m_noEval,
-					m_exit
-				))
+				m_localWorkSize,
+				m_globalWorkSizeMultiplier,
+				m_openclPlatform,
+				0,
+				m_dagLoadMode,
+				m_dagCreateDevice,
+				m_noEval,
+				m_exit
+			)) {
+				stop_io_service();
```

Comment:
This `stop_io_service()` seems to be a bad pattern. Is there a way to start the service only when needed?

Long term, I have better way of handling list of devices.

Reply:
Actually I start the io_service at MinerCLI constructor thus I need to stop it whenever MinerCLI exits

Reply:
Also ... aside from list-devices asio_service is always needed.